packages/telemetry/README.md

```diff
@@ -0,0 +1,278 @@
+})
+
+```
+
+### Browser
+
+To access the browser telemetry singleton use the browser export directly.
+
+```js
+import { telemetry } from '@packages/telemetry/src/browser'
+
+telemetry.init({options})
+```
+
+The browser singleton is also stored on window, in some cases when the telemetry package is included in multiple packages you can use the `attach` method to retrieve and setup the singleton from the instance saved on window.
+
+```js
+import { telemetry } from '@packages/telemetry/src/browser'
+
+telemetry.attach()
+```
+
+The browser telemetry instance only supports the websocket exporter and the websocket must be attached to the exporter for spans to be sent.
+
+```js
+telemetry.attachWebSocket(ws)
+```
+
+The receiver of the websocket message is responsible for forwarding it on to an OTEL collector.
+
+The `OTLPTraceExporterCloud` exporter has a handy send method you can use to forward on the telemetry requests.
+
+```js
+const { OTLPTraceExporterIPC } = require('@packages/telemetry')
+
+(telemetry.exporter()as OTLPTraceExporterCloud)?.send(data, () => {}, (err) => {
+  debug('error exporting telemetry data %s', err)
+})
+
+```
+
+### Shutting down
+
+To ensure all telemetry data is sent, you must shut down any telemetry instances before exiting a process, or refreshing the browser.
+
+```js
+const { telemetry } = require('@packages/telemetry')
+
+await telemetry.shutdown()
+```
+
+## Usage
+
+To access the telemetry instance simply require or import the file.
+
+Node:
+
+```js
+const { telemetry } = require('@packages/telemetry')
+```
+
+Browser:
+
+```js
+import { telemetry } from '@packages/telemetry/src/browser'
+```
+
+### Spans
+
+Spans are the backbone of our telemetry system. At a basic level they measure the time between two points
+
+#### Start
+
+Starting a span indicates the start of a point that you with to time.
+
+We've created a method for starting spans ourselves to allow us to start active spans without wrapping the observed code in a callback. We can also find previously started spans and stop them in other events.
+
+```js
+const { telemetry } = require('@packages/telemetry')
+
+// The returned span is an otel span with all it's properties.
+const span = telemetry.start({name: 'span'})
+
+eventToTime()
+
+// Attributes can be placed on a span to inform parameters that may influence timing, such as a filename.
+span?.setAttributes({
+  fileName: 'filename',
+})
+
+// Spans must be ended for them to be sent to the telemetry endpoint.
+span?.end()
+```
+
+#### Get
+
+Sometimes spans aren't able to be closed in the same function that they're opened, or there are multiple paths that could close the span.
+
+**Warning!** duplicate span names are not handled and the original span will no longer be able to be found.
+
+```js
+const { telemetry } = require('@packages/telemetry')
+
+// The returned span is an otel span with all it's properties.
+const span = telemetry.getSpan('span')
```

Comment:
Is it worth mentioning that the span may not exist if the no op telemetry is used or if the span was never created, which makes the implied `span?.end()` a bit clearer?

Reply:
done